ocs_ci/ocs/external_ceph.py

```diff
@@ -0,0 +1,904 @@
+    def role_to_node_mapping(self):
+        """
+        Given a role we should be able to get the corresponding
+        CephNode object
+
+        """
+        self.role_to_node = {}
+
+
+class RolesContainer(object):
+    """
+    Container for single or multiple node roles.
+    Can be used as iterable or with equality '==' operator to check if role is present for the node.
+    Note that '==' operator will behave the same way as 'in' operator i.e. check that value is present in the role list.
+    """
+
+    def __init__(self, role='pool'):
+        if isinstance(role, str):
+            self.role_list = [str(role)]
+        else:
+            self.role_list = role if len(role) > 0 else ['pool']
+
+    def __eq__(self, role):
+        if isinstance(role, str):
+            return role in self.role_list
+        else:
+            return all(atomic_role in role for atomic_role in self.role_list)
+
+    def __ne__(self, role):
+        return not self.__eq__(role)
+
+    def equals(self, other):
+        if getattr(other, 'role_list') == self.role_list:
+            return True
+        else:
+            return False
+
+    def __len__(self):
+        return len(self.role_list)
+
+    def __getitem__(self, key):
+        return self.role_list[key]
+
+    def __setitem__(self, key, value):
+        self.role_list[key] = value
+
+    def __delitem__(self, key):
+        del self.role_list[key]
+
+    def __iter__(self):
+        return iter(self.role_list)
+
+    def remove(self, object):
+        self.role_list.remove(object)
+
+    def append(self, object):
+        self.role_list.append(object)
+
+    def extend(self, iterable):
+        self.role_list.extend(iterable)
+        self.role_list = list(set(self.role_list))
+
+    def update_role(self, roles_list):
+        if 'pool' in self.role_list:
+            self.role_list.remove('pool')
+        self.extend(roles_list)
+
+    def clear(self):
+        self.role_list = ['pool']
+
+
+class NodeVolume(object):
+    FREE = 'free'
+    ALLOCATED = 'allocated'
+
+    def __init__(self, status):
+        self.status = status
+
+
+class SSHConnectionManager(object):
+    def __init__(self, ip_address, username, password, look_for_keys=False, outage_timeout=300):
+        self.ip_address = ip_address
+        self.username = username
+        self.password = password
+        self.look_for_keys = look_for_keys
+        self.__client = paramiko.SSHClient()
+        self.__client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        self.__transport = None
+        self.__outage_start_time = None
+        self.outage_timeout = datetime.timedelta(seconds=outage_timeout)
+
+    @property
+    def client(self):
+        return self.get_client()
+
+    def get_client(self):
+        if not (self.__transport and self.__transport.is_active()):
+            self.__connect()
+            self.__transport = self.__client.get_transport()
+
+        return self.__client
+
+    def __connect(self):
```

Comment:
I think we only want one prefixed underscore, not two. Change this for similar variables like `__client`, `__transport`, etc. Note that I'm not talking about the [dunder](https://www.geeksforgeeks.org/dunder-magic-methods-python/) methods like `__init__`, just the "private" variables you've created.

```suggestion
    def _connect(self):
```

Reply:
done